submissions/sung-anthony/scripts/article.js

```diff
@@ -0,0 +1,114 @@
+ the prototype, as that would only be relevant to a single instantiated Article.
+ */
+Article.all = [];
+
+Article.prototype.toHtml = function(scriptTemplateId) {
+  var template = Handlebars.compile((scriptTemplateId).html());
+
+  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+  this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+  this.body = marked(this.body);
+
+  return template(this);
+};
+
+/* NOTE: There are some other functions that also relate to articles across the
+ board, rather than just single instances. Object-oriented programming would
+ call these "class-level" functions, that are relevant to the entire "class"
+ of objects that are Articles. */
+
+/* DONE: This function will take our data, however it is provided,
+ and use it to instantiate all the articles. This code is moved from elsewhere,
+ and encapsulated in a simply-named function for clarity. */
+Article.loadAll = function(dataWePassIn) {
+  dataWePassIn.sort(function(a,b) {
+    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+  });
+  dataWePassIn.forEach(function(ele) {
+    Article.all.push(new Article(ele));
+  });
+};
+
+/* This function below will retrieve the data from either a local or remote
+ source, process it, then hand off control to the View. */
+Article.fetchAll = function() {
+  if (localStorage.hackerIpsum) {
+    /* When our data is already in localStorage:
+     1. We can process it by calling the .loadAll() method (started below),
+     2. Then We can render the index page (using the proper method on the
+        articleView object). */
+
+    // Article.loadAll(//TODO: Process our localStorage!
+    // Tip: Be careful when handling different data types between localStorage!
+    // );
+    $.ajax( {
+      url: '../data/hackerIpsum.json',
+      success: function(data, message, xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        if (eTag !== JSON.parse(localStorage.getItem(('stringedETag')))) {
+          Article.all = [];
+          Article.getData('../data/hackerIpsum.json');
+        }
+      }
+    });
+
+    //TODO: Now call the correct method here that will render the index page.
+  } else {
+    /* TODO: When we don't already have our data, we need to:
+
+     - Retrieve our JSON file with AJAX
+       (which jQuery method is best for this?).
+
+     Now within this method, we can:
+      1. Pass the resulting JSON data into the .loadAll method
+
+      2. Store that same data in localStorage so we can skip the server call next time
+
+      3. And then render the index page (What method was that?) */
+
+    $.ajax( {
+      url: '../data/hackerIpsum.json',
+      success: function(data, message, xhr) {
+        Article.getData('../data/hackerIpsum.json');
+        var eTag = xhr.getResponseHeader('etag');
+        localStorage.setItem('stringedETag', JSON.stringify(eTag));
```

Comment:
localStorage can be accessed like any javascript object
for gets you can do: localStorage.stringedETag
and sets: localStorage.stringedETag = 'myetag'